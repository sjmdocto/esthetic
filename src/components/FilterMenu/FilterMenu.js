import React, {useState} from 'react';
import {
  Modal,
  View,
  Text,
  TouchableWithoutFeedback,
  TouchableOpacity,
} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import ColorChoices from './Option/ColorChoices';
import TypeChoices from './Option/TypeChoices';
import {styles} from './FilterMenu.style';

/**
 * @function FilterMenu
 * @param {*} props
 * @param {boolean} props.visible - sets visibility of menu
 * @param {func} props.onClose - close menu
 * @param {func} props.setFilterColor
 * @param {func} props.setFilterType
 */

const FilterMenu = (props) => {
  /* COLOR FILTER CODE */
  const [selectedColor, setSelectedColor] = useState(0);

  /**
   * Sets menu's selected color to input color,
   * and sets global filter color to input color.
   * Not used by menu, but is passed to Option
   * @function changeSelectedClothing
   * @param {number} color
   * @returns {void}
   */
  const changeSelectedColor = (color) => {
    setSelectedColor(color);
    props.setFilterColor(color);
  };

  /* TYPE FILTER CODE */
  const [selectedClothing, setSelectedClothing] = useState(0);

  /**
   * Sets menu's selected type to input type,
   * and sets global filter type to input type.
   * Not used by menu, but is passed to OptionText
   * @function changeSelectedClothing
   * @param {number} clothing
   * @returns {void}
   */
  const changeSelectedClothing = (clothing) => {
    setSelectedClothing(clothing);
    props.setFilterType(clothing);
  };

  return (
    <Modal
      visible={props.visible}
      transparent={true}
      statusBarTranslucent={true}
      animationType={'fade'}>
      <View style={styles.menuScreen}>
        <View style={styles.menuArea}>
          <View style={styles.menuHeader}>
            <Text style={styles.filtersTitle}>Filters</Text>
          </View>
          <View style={styles.menuBody}>
            <View style={styles.colorsContainer}>
              <View style={styles.categoriesHeader}>
                <Icon name="circle" size={18} color={'white'} />
                <Text style={styles.categoriesText}>Colors</Text>
              </View>
              <ColorChoices
                style={styles.optionsContainer}
                currentColor={selectedColor}
                selectColor={changeSelectedColor}
              />
            </View>
            <View style={styles.clothingContainer}>
              <View style={styles.categoriesHeader}>
                <Icon name="circle" size={18} color={'white'} />
                <Text style={styles.categoriesText}>Clothing</Text>
              </View>
              <TypeChoices
                style={styles.optionsContainer}
                currentType={selectedClothing}
                selectType={changeSelectedClothing}
              />
            </View>
            <View style={styles.doneContainer}>
              <TouchableOpacity
                onPress={props.onClose}
                style={styles.doneButton}
                accessible={true}
                accessibilityLabel={'Done button'}>
                <Text style={styles.doneText}>Done</Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
        <TouchableWithoutFeedback onPress={props.onClose}>
          <View
            style={styles.nonMenuArea}
            accessible={true}
            accessibilityLabel={'Outside of Filter Menu area'}
            accessibilityHint={'Closes Filter Menu'}
          />
        </TouchableWithoutFeedback>
      </View>
    </Modal>
  );
};

export default FilterMenu;
