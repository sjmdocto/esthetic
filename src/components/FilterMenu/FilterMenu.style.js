import {StyleSheet, Platform, StatusBar} from 'react-native';
import theme from '../../styles/theme.style';

export const styles = StyleSheet.create({
  menuScreen: {
    flexDirection: 'row',
    flex: 1,
    // if android, can use StatusBar.currentHeight
    // else, for iOS, has to be set manually to 20
    // PROBLEM: dealing with android's with notches
    // and iPhone's w/o notches
    paddingTop: Platform.OS === 'android' ? StatusBar.currentHeight : 0,
  },
  menuArea: {
    backgroundColor: theme.EST_ORANGE_TRANSP,
    flex: 9,
    color: 'white',
    // only looks good on iPhones w/notches
    paddingTop: Platform.OS === 'android' ? 0 : 45,
    paddingBottom: Platform.OS === 'android' ? 0 : 10,
  },
  menuHeader: {
    borderBottomWidth: 1,
    borderBottomColor: 'white',
    //height: '8%',
    flex: 8,
  },
  filtersTitle: {
    color: 'white',
    paddingTop: '3%',
    paddingLeft: '5%',
    fontWeight: 'bold',
    fontSize: 30,
  },
  menuBody: {
    paddingTop: '5%',
    paddingLeft: '5%',
    flex: 100,
  },
  colorsContainer: {
    flexGrow: 1,
  },
  categoriesHeader: {
    flexDirection: 'row',
  },
  clothingContainer: {
    flexGrow: 1,
  },
  optionsContainer: {
    flexGrow: 1,
    justifyContent: 'space-evenly',
    paddingLeft: '20%',
  },
  categoriesText: {
    paddingLeft: 5,
    color: 'white',
    fontSize: 16,
  },
  doneContainer: {
    paddingBottom: '5%',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  doneButton: {
    //borderWidth: 1,
    width: 50,
    alignItems: 'center',
  },
  doneText: {
    fontSize: 16,
    fontWeight: 'bold',
    color: 'white',
  },
  nonMenuArea: {
    flex: 4,
  },
});
