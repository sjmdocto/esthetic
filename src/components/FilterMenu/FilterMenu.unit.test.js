import React from 'react';
import FilterMenu from './FilterMenu';
import {render, fireEvent} from 'react-native-testing-library';

describe('FilterMenu test suite:', () => {
  const mockOnClose = jest.fn().mockName('onClose');
  const mockSetFilterColor = jest.fn().mockName('setFilterColor');
  const mockSetFilterType = jest.fn().mockName('setFilterType');

  test('Done button is functional.', () => {
    const {getByLabelText} = render(
      <FilterMenu visible={true} onClose={mockOnClose} />,
    );
    fireEvent.press(getByLabelText('Done button'));
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('Pressing outside of filterMenu calls onClose.', () => {
    const {getByLabelText} = render(
      <FilterMenu visible={true} onClose={mockOnClose} />,
    );
    fireEvent.press(getByLabelText('Outside of Filter Menu area'));
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('Color Filter buttons are functional.', () => {
    const {getByLabelText} = render(
      <FilterMenu visible={true} setFilterColor={mockSetFilterColor} />,
    );
    fireEvent.press(getByLabelText('Blue filter button'));
    expect(mockSetFilterColor).toHaveBeenCalled();
  });

  test('Type Filter buttons are functional.', () => {
    const {getByLabelText} = render(
      <FilterMenu visible={true} setFilterType={mockSetFilterType} />,
    );
    fireEvent.press(getByLabelText('Hoodies filter button'));
    expect(mockSetFilterType).toHaveBeenCalled();
  });

  test('FilterMenu renders properly.', () => {
    const {toJSON} = render(<FilterMenu visible={true} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
