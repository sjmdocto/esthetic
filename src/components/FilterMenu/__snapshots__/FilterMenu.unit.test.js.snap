// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu test suite: FilterMenu renders properly. 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingTop": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(227, 131, 4, 0.96)",
          "color": "white",
          "flex": 9,
          "paddingBottom": 10,
          "paddingTop": 45,
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomColor": "white",
            "borderBottomWidth": 1,
            "flex": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 30,
              "fontWeight": "bold",
              "paddingLeft": "5%",
              "paddingTop": "3%",
            }
          }
        >
          Filters
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 100,
            "paddingLeft": "5%",
            "paddingTop": "5%",
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï„·
            </Text>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "paddingLeft": 5,
                }
              }
            >
              Colors
            </Text>
          </View>
          <View
            style={
              Object {
                "flexGrow": 1,
                "justifyContent": "space-evenly",
                "paddingLeft": "20%",
              }
            }
          >
            <View
              accessibilityLabel="Black filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Black
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="White filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  White
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Grey filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Grey
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Red filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Red
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Orange filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Orange
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Yellow filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Yellow
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Green filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Green
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Blue filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Blue
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Purple filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Purple
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Multicolored filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Multicolored
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï„·
            </Text>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "paddingLeft": 5,
                }
              }
            >
              Clothing
            </Text>
          </View>
          <View
            style={
              Object {
                "flexGrow": 1,
                "justifyContent": "space-evenly",
                "paddingLeft": "20%",
              }
            }
          >
            <View
              accessibilityLabel="Outerwear filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Outerwear
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Formal Wear filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Formal Wear
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Sweaters filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Sweaters
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Hoodies filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Hoodies
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Button Ups filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Button Ups
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Tees filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Tees
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Pants filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Pants
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Shorts filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Shorts
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Activewear filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Activewear
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Shoes filter button"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "width": 100,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Shoes
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": "5%",
            }
          }
        >
          <View
            accessibilityLabel="Done button"
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "opacity": 1,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Done
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityHint="Closes Filter Menu"
      accessibilityLabel="Outside of Filter Menu area"
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 4,
        }
      }
    />
  </View>
</Modal>
`;
